.github/workflows/build-and-test-actuated.yml

```diff
@@ -0,0 +1,77 @@
+name: build-and-test
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+  merge_group:
+  pull_request:
+env:
+  TEST_RESULTS: testbed/tests/results/junit/results.xml
+  # Make sure to exit early if cache segment download times out after 2 minutes.
+  # We limit cache download as a whole to 5 minutes.
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
+  GOPROXY: https://goproxy1.cncf.selfactuated.dev,direct
+
+# Do not cancel this workflow on main. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16616
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  actuated-unittest-matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        group:
+          - receiver-0
+          - receiver-1
+          - receiver-2
+          - receiver-3
+          - processor
```

Comment:
Either this PR or https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32526 needs a rebase & fix depending on which one gets in first.

cc @jmacd

Reply:
@songy23 merged #32526 since it's been ready to merge. will update this pr

Reply:
Fixed